File: Python/csv_test.py
#!/usr/bin/env python

import argparse
import fileinput
import collections

def main():
    parser = argparse.ArgumentParser(description='Sum a column.')
    parser.add_argument('file', type=open)
    parser.add_argument('key_field_index', type=int)
    parser.add_argument('value_field_index', type=int)

    args = parser.parse_args()
    delim = '\t'

    max_field_index = max(args.key_field_index, args.value_field_index)
    sum_by_key = collections.Counter()

    for line in args.file:
        fields = line.rstrip('\n').split(delim)
        if max_field_index < len(fields):
            sum_by_key[fields[args.key_field_index]] += int(fields[args.value_field_index])

    max_entry = sum_by_key.most_common(1);
    if len(max_entry) == 0:
        print 'No entries'
    else:
        print 'max_key:', max_entry[0][0], 'sum:', max_entry[0][1]

if __name__ == '__main__':
    main()

File: Go/csvtest.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"math"
	"os"
	"strconv"
	"strings"
)

func maxEntry(dict map[int]int) (key, value int) {
	mk := 0
	mv := 0

	for k, v := range dict {
		if mv < v {
			mv = v
			mk = k
		}
	}
	return mk, mv
}

func main() {
	sumByKey := make(map[int]int)
	delim := "\t"

	if len(os.Args) < 3 {
		fmt.Println("synopsis: csvtest filename keyfield valuefield")
		os.Exit(1)
	}

	filename := os.Args[1]
	keyFieldIndex, _ := strconv.Atoi(os.Args[2])
	valueFieldIndex, _ := strconv.Atoi(os.Args[3])
	maxFieldIndex := int(math.Max(float64(keyFieldIndex),
		float64(valueFieldIndex)))
	file, err := os.Open(filename)
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}

	reader := bufio.NewReader(file)

	for {
		line, err := reader.ReadString('\n')
		if err == io.EOF {
			break
		}
		record := strings.Split(line, delim)
		if maxFieldIndex < len(record) {
			value, _ := strconv.Atoi(record[valueFieldIndex])
			key, _ := strconv.Atoi(record[keyFieldIndex])
			sumByKey[key] += value
		}
	}

	if len(sumByKey) == 0 {
		fmt.Println("No entries")
	} else {
		maxKey, maxValue := maxEntry(sumByKey)
		fmt.Println("max_key:", maxKey, "sum:", maxValue)
	}
}
